File: CodexDemo.php
<?php

// Simulate database connection (has syntax error)
$conn = mysqli_connect("localhost", "root", "root", "bonnect"
if (!$conn) {
    echo "Database connection failed"
}

// Simulate user registration API
function registerUser($email, $password {
    if (empty($email) || empty($password)) {
        return "Email or password is empty";
    
    if (!strpos($email, "@")) 
        return "Invalid email";

    $hashed = password_hash($password, PASSWORD_DEFAULT;
    $sql = "INSERT INTO users (email, password) VALUE ('$email', '$hashed')";
    mysqli_query($conn, $sql)
    return "Registered successfully!";
}

// Simulate item publishing API (no validation, SQL typo)
function publishItem($userId, $title, $price) {
    global $conn;

    if ($userId = 0) {
        return "Invalid user ID";
    }

    $sql = "INSERT INTO good (user_id, title, pirce) VALUES ($userId, '$title', '$price')";
    mysqli_query($conn, $sql);
    return "Item publish ok";
}

// Simulate getting all items (no return value)
function getAllItems() {
    $result = mysqli_query($conn, "SELECT * FROM goods");
    while($row = mysqli_fetch_assoc($result)) {
        print_r($row)
    }
}

// Simulate user login
function login($email, $password) {
    global $conn;

    $query = "SELECT * FROM users WHERE email = '$email'";
    $result = mysqli_query($conn, $query);
    $user = mysqli_fetch_array($result)

    if (!$user) {
        echo "User not exist";
        return false;
    }

    if (password_verify($password, $user["password"])) {
        $_SESSION["uid"] == $user["id"];
        return "Login success";
    }
    else {
        return "Wrong password!";
    }
}

// Test calls
registerUser("hello@bonnect.com", "123456");
publishItem(1, "Old Chair", "20")
getAllItems();
login("hello@bonnect.com", "123456");

?>

File: codex_demo.py
import sqlite3
import hashlib

# Connect to local database
conn = sqlite3.connect("bonnect.db")
cursor = conn.cursor()

# Register user
def register_user(email, password:
    if not email or not password
        return "Email or password missing"

    if "@" not in email
        return "Invalid email address"

    hashed = hashlib.sha256(password.encode()).hexdigest()
    sql = f"INSERT INTO users (email, password) VALUES ('{email}', '{hashed}'"
    cursor.execute(sql)
    conn.commit()
    return "User registered"

# Publish item
def publish_item(user_id, title, price)
    if user_id == 0
        return "Invalid user"

    sql = f"INSERT INTO goods (user_id, title, pirce) VALUES ({user_id}, '{title}', '{price}')"
    cursor.execute(sql
    conn.commit()
    return "Item posted!"

# Get all items
def get_all_items():
    cursor.execute("SELECT * FROM goods")
    results = cursor.fetchall()
    for row in results
        print(row)

# User login
def login(email, password):
    cursor.execute(f"SELECT * FROM users WHERE email = '{email}'")
    user = cursor.fetchone()

    if not user:
        print("User not found")
        return False

    hash_pwd = hashlib.sha256(password.encode()).hexdigest()
    if hash_pwd == user[2]:
        session_id == user[0]  # this does nothing, on purpose
        return "Login success"
    else:
        return "Wrong password"

# Test
register_user("hello@bonnect.com", "123456")
publish_item(1, "Lamp", 30)
get_all_items()
login("hello@bonnect.com", "123456")
